ddcloud/resource_server.go

```diff
@@ -488,6 +512,44 @@ func resourceServerUpdate(data *schema.ResourceData, provider interface{}) error
 		}
 	}
 
+	if data.HasChange(resourceKeyServerPowerState) {
+		log.Printf("Server power state change has been detected.")
+		powerState := propertyHelper.GetOptionalString(resourceKeyServerPowerState, false)
+
+		if powerState != nil {
+			switch strings.ToLower(*powerState) {
+			case "start":
+				err = serverStart(providerState, serverID)
+			case "shutdown":
+				err = serverShutdown(providerState, serverID)
+			case "shutdown-hard":
+				err = serverPowerOff(providerState, serverID)
+			case "disabled", "autostart":
```

Comment:
Thanks, BTW - in all the time I've used Go it never occurred to me that you could do this!

Reply:
This is the first time i've written go.
Coming from a c#/java and python background. I looked ways to my work flow. Therethere maybe some code that shouldnt be there :)

Like i used break for an empty case. I know go doesnt require it. Just felt natural and for a non-programmer self documenting ;)

Reply:
Go's not my daily driver (come from a similar background, and mainly use Go for extending stuff built in Go or things that need cross-compiling and static linking) but I figured I knew most of the ins and outs of the language (there's not a lot to learn really). Just goes to show you, you can learn something new with every PR :)

PS. Agree about `break`